sys/conf/options

```diff
@@ -1019,6 +1019,7 @@ LINDEBUGFS
 
 # CHERI
 CPU_CHERI	opt_global.h
+CHERI_PURECAP_KERNEL		opt_global.h
```

Comment:
Can we keep this option out of dev?

Reply:
Not sure, it is currently used to conditionally compile a bunch of files. We generally have most of the changes in common files but for MIPS I duplicated a bunch of assembly files because everything changed.

Reply:
If we need it for files we can change it to opt_dontuse.h to prevent use.

Reply:
And for those cases should it not just be `CHERI_PURECAP`? The `_KERNEL` is redundant.

Reply:
It is similar to CPU_CHERI in that they shouldn't be used in #ifdef's, only for sys/conf/files.  I'm not sure if we've actually cleaned up CPU_CHERI enough to move it to opt_dontuse.h, but CHERI_PURECAP_KERNEL should be clean enough to move.

I don't really care strongly what the name is, but it would be nice if CPU_CHERI and CHERI_PURECAP were a bit closer in names, perhaps CPU_CHERI should be CHERI_HYBRID instead?

Reply:
CPU_CHERI is _meant_ to be somewhat akin to CPU_QEMU_MALTA and CPU_BERI. It's used almost nowhere outside of sys/mips. But even in the current situation it's more like `__has_feature(capabilities)`, whereas CHERI_HYBRID would not be true for CHERI_PURECAP.

Reply:
i.e. really we need to split it out into a genuine `__has_feature(capabilities)` option for files and a CPU_CHERI to mean CHERI-MIPS (though maybe that can be dropped and just use CPU_BERI?).

Reply:
Hmm, looking at sys/conf/files, CHERI_HYBRID doesn't work indeed.  CPU_CHERI really means either hybrid or purecap, it is the __has_feature() equivalent you mentioned.  CPU_CHERI is probably fine for that name-wise as CHERI itself is not BERI specific.  I guess I don't have a better name.  At least CHERI_PURECAP is shorter than `__CHERI_PURE_CAPABILITY__`.

Reply:
This still needs moving to opt_dontuse.h.